deeplake/core/tensor.py

```diff
@@ -591,16 +591,19 @@ def __getitem__(
             item = replace_ellipsis_with_slices(item, self.ndim)
         if not is_iteration and isinstance(item, int):
             indexing_history = self._indexing_history
-            if len(indexing_history) == 2:
-                a, b = indexing_history
-                if item - b == b - a:
+            if len(indexing_history) == 10:
```

Comment:
can we refactor this out? The code is repeated in both tensor and dataset classes except for the warning message